src/app/store/router-store/custom-router-serializer.ts

```diff
@@ -0,0 +1,32 @@
+import { Params, RouterStateSnapshot } from '@angular/router';
+import { RouterStateSerializer } from '@ngrx/router-store';
+
+export interface RouterStateUrl {
+  url: string;
+  params: Params;
+  queryParams: Params;
+}
+
+export class CustomRouterSerializer implements RouterStateSerializer<RouterStateUrl> {
+  serialize(routerState: RouterStateSnapshot): RouterStateUrl {
+    let route = routerState.root;
+    let params = {};
+
```

Comment:
This serializer runs on all the routing events and allows us to get the params from all the route segments for digesting into the router-store.